### DISCUSSION TITLE
Error- 'TData' is defined but never used.eslint@typescript-eslint/no-unused-vars

### DISCUSSION THREAD

**Discussion by kcalderon37** [TanStack/table#5222] [Q&A]
Hi,

I'm encountering issues while declaring an interface and receiving errors related to TData and TValue. I followed the same structure as the example below:

import '@tanstack/react-table'

declare module '@tanstack/table-core' {
  interface ColumnMeta<TData extends RowData, TValue> {
    foo: string
  }
}

The issues are like:

"All type parameters are unused.ts(6205)
'TData' is defined but never used.eslint[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)
'TData' is defined but never used.eslint[no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)
(type parameter) TData in ColumnMeta<TData extends unknown, TValue>"

the only solution for now is adding comment:
  // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars

Im using version: 8.10.7.  Is anyone else experiencing the same issue?

**Comment by kippo** [comment]
Yes I'm also facing this issue and like you I've had to ignore the issue. It would be nice if we could get this fixed.

**Comment by OmarZeidan** [comment]
I am wondering if any solution to this instead of adding the `// eslint-disable-next-line @typescript-eslint/no-unused-vars`

**Comment by erlandsona** [comment]
Upvoting same for me.

**Comment by GrahamWilsdon** [comment]
`TData` + `TValue`
`defined but never used.eslint[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)`

Still persisting in `8.20.5` from [following docs advice ](https://tanstack.com/table/latest/docs/api/core/column-def#meta)

**Comment by nilsso** [comment]
I find that `// eslint-disable-next-line @typescript-eslint/no-unused-vars` doesn't hide the warning in neovim (I think because it's a hint and not an error?) so I just declaring a dummy `never & TData & TValue` attribute to get rid of it.
```typescript
declare module '@tanstack/table-core' {
  interface ColumnMeta<TData extends RowData, TValue> {
    // ...
    _?: never & TData & TValue
  }
}
```
